mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/scheduler/ResourceClusterAwareSchedulerActor.java

```diff
@@ -120,6 +135,58 @@ public Receive createReceive() {
             .build();
     }
 
+    private void onBatchScheduleRequestEvent(BatchScheduleRequestEvent event) {
+        log.info("Received batch schedule request event: {}", event);
+
+        // If the size of the batch request is 1 we'll handle it as the "old" schedule request
+        if (event.getRequest().getScheduleRequests().size() == 1) {
```

Comment:
any reason/upside to keeping the 1 node case special?

Reply:
Good question. If we're talking about actions like autoscaling or resubmitting a worker, scheduling requests will always going to involve one worker at the time. In these situations, we'd rather use the `ScheduleRequestEvent` lifecycle, not the `BatchScheduleRequestEvent`. The reason is that `ScheduleRequestEvent` lets us try again at the worker level. This means setting up timers for each worker (using their worker-id), instead of using timers for the entire job (with job-id) like we would in `BatchScheduleRequestEvent`, which will collide.